runtime/src/main/java/org/corfudb/runtime/DistributedCompactor.java

```diff
@@ -0,0 +1,561 @@
+                    .setClientName(this.clientName)
+                    .setTableSize(corfuTable.size())
+                    .setTimeTaken(System.currentTimeMillis() - tableCkptStartTime)
+                    .build();
+        } catch (Exception e) {
+            log.error("Unable to checkpoint table: {}, e: {}", tableName, e);
+            returnStatus = CheckpointingStatus.newBuilder()
+                    .setStatus(CheckpointingStatus.StatusType.FAILED)
+                    .setClientName(this.clientName)
+                    .setTableSize(corfuTable.size())
+                    .setTimeTaken(System.currentTimeMillis() - tableCkptStartTime)
+                    .build();
+        }
+        livenessUpdater.notifyOnSyncComplete();
+        return returnStatus;
+    }
+
+    private CorfuTable<CorfuDynamicKey, OpaqueCorfuDynamicRecord> openTable(TableName tableName,
+                                                                            ISerializer serializer,
+                                                                            CorfuRuntime rt) {
+        log.info("Opening table {} in namespace {}", tableName.getTableName(), tableName.getNamespace());
+        SMRObject.Builder<CorfuTable<CorfuDynamicKey, OpaqueCorfuDynamicRecord>> corfuTableBuilder =
+                rt.getObjectsView().build()
+                        .setTypeToken(new TypeToken<CorfuTable<CorfuDynamicKey, OpaqueCorfuDynamicRecord>>() {
+                        })
+                        .setStreamName(getFullyQualifiedTableName(tableName.getNamespace(), tableName.getTableName()))
+                        .setSerializer(serializer)
+                        .addOpenOption(ObjectOpenOption.NO_CACHE);
+        if (persistedCacheRoot == null || persistedCacheRoot.equals(EMPTY_STRING)) {
+            log.warn("Table {}::{} should be opened in disk-mode, but disk cache path is invalid",
+                    tableName.getNamespace(), tableName.getTableName());
+        } else {
+            final String persistentCacheDirName = String.format("compactor_%s_%s",
+                    tableName.getNamespace(), tableName.getTableName());
+            final Path persistedCacheLocation = Paths.get(persistedCacheRoot).resolve(persistentCacheDirName);
+            final Supplier<StreamingMap<CorfuDynamicKey, OpaqueCorfuDynamicRecord>> mapSupplier =
+                    () -> new PersistedStreamingMap<>(
+                            persistedCacheLocation, PersistedStreamingMap.getPersistedStreamingMapOptions(),
+                            serializer, rt);
+            corfuTableBuilder.setArguments(mapSupplier, ICorfuVersionPolicy.MONOTONIC);
+        }
+        return corfuTableBuilder.open();
+    }
+
+    /**
+     * Checkpoint all tables already opened by the current JVM to save on reads & memory.
+     *
+     * @return positive count on success and negative value on failure
+     */
+    public synchronized int checkpointOpenedTables() {
```

Comment:
Is this synchronized keyword required? From what I can tell, instances of this class are only accessed by one thread.